python/src/iceberg/expressions/base.py

```diff
@@ -138,6 +75,24 @@ def __invert__(self) -> "BooleanExpression":
         ...
 
 
+class BoundPredicate(ABC):
+    def __init__(self, term, literal):
```

Comment:
Having a literal in `BoundPredicate` doesn't quite work, since different bound predicates will have different literal counts. See my note below. You may just want to leave this as an empty class for now.